src/lib.rs

```diff
@@ -38,18 +38,18 @@ impl Lehmer {
         Self { code }
     }
 
-    pub fn to_permutation(mut self) -> Vec<u64> {
+    pub fn as_permutation(&mut self) -> Vec<u64> {
         let n = self.code.len() as u64;
         let mut sequence: Vec<u64> = (0..n).collect();
 
         for d in &mut self.code {
             *d = sequence.remove(*d as usize);
```

Comment:
The risk here is that we are mutating `self.code` in place and then returning an owned cope of it: does that mean it's not possible to call `as_permutation` on the same `Lehmer` twice?

Reply:
This makes me think that these really _are_ `to_` methods but it's the mutable reference to `self` that is the problem here. Let me try getting rid of that...